src/Avalonia.Base/Reactive/BindingValueExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Reactive.Subjects;
+using Avalonia.Data;
+
+#nullable enable
+
+namespace Avalonia.Reactive
+{
+    public static class BindingValueExtensions
+    {
+        public static IObservable<BindingValue<T>> ToBindingValue<T>(this IObservable<T> source)
+        {
+            source = source ?? throw new ArgumentNullException(nameof(source));
+            return new BindingValueAdapter<T>(source);
+        }
+
+        public static ISubject<BindingValue<T>> ToBindingValue<T>(this ISubject<T> source)
+        {
+            source = source ?? throw new ArgumentNullException(nameof(source));
+            return new BindingValueSubjectAdapter<T>(source);
+        }
+
+        public static IObservable<object> ToUntyped<T>(this IObservable<BindingValue<T>> source)
```

Comment:
I think those both would need to return `IObservable<object?>` to satisfy nullability requirements and get rid of the warnings.